### ISSUE TITLE
Preparing for 1.0

### ISSUE THREAD

**Issue by ltjessem** [DavidMStraub/homeassistant-homeconnect#76]
First off, congratulations on the inclusion in HA core! 
Should those of us using homeconnect either manually installed or through HACS manually rename or remove the custom component before upgrading HA when the time comes? Or does HACS automatically rename it with _beta? 

The enhanced permissions that can be applied for sound promising, has this been done? If not hopefully you can keep pushing for this to happen.

**Comment by DavidMStraub** [comment]
Thanks!

> Should those of us using homeconnect either manually installed or through HACS manually rename or remove the custom component before upgrading HA when the time comes? Or does HACS automatically rename it with _beta?

We have to remove it. The old custom integration is named `homeconnect`, the official integration `home_connect` and the current version of the custom one (that could also be used with HACS) `home_connect_beta`. In principle they could coexist, but it doesn't make sense to use them with the same Home Connect account. It might trigger API rate limits.

> The enhanced permissions that can be applied for sound promising, has this been done? If not hopefully you can keep pushing for this to happen.

No, this has to be done by the HA Core team, I will ask them.

**Comment by h4nc** [comment]
What is the correct way to switch from the custom integration to the HA core integration? I'd like to avoid double or unused entities? 

> The enhanced permissions that can be applied for sound promising, has this been done?

Also what is this about? Could you clarify that please?

**Comment by h4nc** [comment]
What is the correct way to switch from the custom integration (v0.3) to the HA core integration? Would be great to know this before updating to 0.110. I'd like to avoid double or unused entities.

> The enhanced permissions that can be applied for sound promising, has this been done?

Also what is this about? Could you clarify that please?

**Comment by DavidMStraub** [comment]
> What is the correct way to switch from the custom integration (v0.3) to the HA core integration? Would be great to know this before updating to 0.110. I'd like to avoid double or unused entities.

I haven't done it myself, but you can look at #85 #79 for possible pitfalls.

I would say:

1. Remove the integration in the UI
2. Uninstall the HACS component or delete the custom component
3. Change the `configuration.yaml`
4. Restart HA
5. Add the new integration in the UI

**Comment by DavidMStraub** [comment]
> Also what is this about? Could you clarify that please?

You can't switch on on oven with a normal developer account. If instead HA users can access the API through a developer account owned by Nabu Casa, they can ask Home Connect to grant them such enhanced permissions. Paulus wrote me that this might happen.

**Comment by h4nc** [comment]
Thanks I will try that. 

> You can't switch on on oven with a normal developer account. 

You mean turning on something like a heating program, right? Because the oven switch I got with this component is able to turn on the "UI" of the oven (performing that starting noice). 
So I guess those enhanced permissions are about heat programs?

**Comment by DavidMStraub** [comment]
Exactly. Yes, I can also make my oven turn on, but not start any program.

**Comment by h4nc** [comment]
Successfully updated to the core version. I noticed that the elapsed_program_time sensor is gone. Was it removed?

BTW: HACS wasn't able to delete the folder in custom_components by itself. Maybe because the current version has a new name (beta).

**Comment by h4nc** [comment]
Successfully updated to the core version. 
As a site note HACS wasn't able to delete the folder in custom_components by itself. Maybe because the current version has a new name (beta). 

BTW: I noticed that the elapsed_program_time sensor is gone. Found that it was removed [here](https://github.com/DavidMStraub/homeassistant-homeconnect/commit/8a3017302f52970551c4da35a17aa057815849e5). May I asked why it was removed?

**Comment by DavidMStraub** [comment]
Yes. It was requested as part of the PR to convert the elapsed and remaining time sensors to timestamp sensors, which makes absolute sense and was easy to implement for the remaining time, since it can just be cleared after the timestamp has passed; but for the elapsed time, it needs to be cleared when the program finishes and this would need some improved logic (#5) to make work reliably, so before delaying the PR even further I just removed it. (I also don't find it super useful personally...)

**Comment by DavidMStraub** [comment]
Oh, and I think this issue can be closed.

**Comment by h4nc** [comment]
I understand thanks. 

I'd like to add that I tried to reinstall v0.3 to see if the reinstalled integrations is able to delete the folder. But the installation failed, it seems like HACS isn't able to install v0.3 any more for some reason (maybe you know why?). In the end I deleted the folder manually. 

I understand that being able to install old versions isn't that important. However if it's not possible to install it in hacs, it might be better to remove them from the version list in hacs (if possible).

**Comment by h4nc** [comment]
I understand, thanks.

I'd like to add that I tried to reinstall v0.3 to see if the reinstalled integrations is able to delete the folder. But the installation failed, it seems like HACS isn't able to install v0.3 any more for some reason (maybe you know why?). In the end I deleted the folder manually.

I understand that being able to install old versions isn't that important. However if it's not possible to install it in HACS, it might be better to remove them from the version list in HACS (if possible).
